nexus/src/db/queries/external_ip.rs

```diff
@@ -183,9 +182,29 @@ const MAX_PORT: i32 = u16::MAX as _;
 /// (even though we send the start/stop to the database as `i32`s). We need to
 /// cast it on the way out of the database, so that Diesel can correctly
 /// deserialize it into an `i32`.
+///
+/// Pool restriction
+/// ----------------
+///
+/// Clients may optionally request an external address from a specific IP Pool.
+/// If they don't provide a pool, the query is restricted to all IP Pools
+/// available to their project (not reserved for a different project). In the
+/// first case, the restriction is:
+///
+/// ```sql
+/// pool_id = <pool_id>
```

Comment:
Do we want/need to add a check that the requested pool is also either reserved for this Instance's Project or not reserved to any Project at all?

Reply:
Hmm, yeah we can do that. I avoided anything like that since it'll change when we add project-scoped endpoints for viewing pools. But I'll add now, thanks!

Reply:
Done in bb936163